pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherSingleActiveConsumer.java

```diff
@@ -366,6 +377,30 @@ protected void readMoreEntries(Consumer consumer) {
         }
     }
 
+    private boolean shouldPauseOnAckStatePersist() {
+        if (!((PersistentTopic) subscription.getTopic()).isDispatcherPauseOnAckStatePersistentEnabled()) {
+            return false;
+        }
+        if (cursor == null) {
+            return true;
+        }
+        if (!cursor.isMetadataTooLargeToPersist()) {
+            return false;
+        }
+        // The cursor state is too large to persist, let us check whether the read is a replay read.
+        Range<PositionImpl> lastIndividualDeletedRange = cursor.getLastIndividualDeletedRange();
+        if (lastIndividualDeletedRange == null) {
+            // lastIndividualDeletedRange is null means the read is not replay read.
+            return true;
+        }
+        // If read position is less than the last acked position, it means the read is a replay read.
+        PositionImpl lastAckedPosition = lastIndividualDeletedRange.upperEndpoint();
+        Position readPosition = cursor.getReadPosition();
+        boolean readPositionIsSmall =
+                lastAckedPosition.compareTo(readPosition.getLedgerId(), readPosition.getEntryId()) > 0;
+        return !readPositionIsSmall;
+    }
+
```

Comment:
Hmmm. For the single active consumer subscription. Practically, I think we don't need to consider the ack state persistence because it should be a misuse. Maybe we can skip the single active consumer subscription right now. And add it later if we really need it, based on the actual needs.

Reply:
Sure, reverted the change of `PersistentDispatcherSingleActiveConsumer`